packages/x-charts/src/internals/plugins/featurePlugins/useChartHighlight/useChartHighlight.ts

```diff
@@ -1,31 +1,39 @@
+import { useAssertModelConsistency } from '@mui/x-internals/useAssertModelConsistency';
 import useEventCallback from '@mui/utils/useEventCallback';
 import useEnhancedEffect from '@mui/utils/useEnhancedEffect';
 import { ChartPlugin } from '../../models';
 import { HighlightItemData, UseChartHighlightSignature } from './useChartHighlight.types';
 
-export const useChartHighlight: ChartPlugin<UseChartHighlightSignature> = ({
-  store,
-  params,
-  models,
-}) => {
+export const useChartHighlight: ChartPlugin<UseChartHighlightSignature> = ({ store, params }) => {
+  useAssertModelConsistency({
+    componentName: 'Chart',
+    propName: 'highlightedItem',
+    controlled: params.highlightedItem,
+    defaultValue: null,
+  });
+
   useEnhancedEffect(() => {
-    store.update((prevState) => ({
-      ...prevState,
-      highlight: {
-        ...prevState.highlight,
-        item: models.highlightedItem.value,
-      },
-    }));
-  }, [store, models.highlightedItem.value]);
+    store.update((prevState) =>
+      prevState.highlight.item === params.highlightedItem
```

Comment:
This equality check is checking objects and will probably cause issues.

Reply:
What issues do you think it might cause? 

We're only updating the `state.highlight.item` here, so if the new value is the same as the old one, it's safe to skip, isn't it?

Reply:
Yes, params is a user input. If `<Component highlightedItem={{id: 1}}` the object will be a new one on each render, which will trigger this hook, which will update the state.